### PR TITLE
Force ID fields to be numeric when JSON-encoded

### PR DESCRIPTION
**Pull request by s-nez**
The DataDog agent requires all ID fields to be properly numeric (they can't be quoted as strings, even if the content is only numbers). This change converts the ID fields into Math::BigInt unless they are already objects (if they are objects, they either are already Math::BigInt or should handle serialization on their own).

### CODE REVIEW

**Comment by vanHoesel** [comment]
That is weird, this should have been fixed in:
63b4b669f59783e5e4d4c7b992785e9af3fbdef5

**Comment by vanHoesel** [comment]
Right, let me think a bit...

these ID's SHOULD already be a `InstanceOf['Math::BigInt']` object, as stated in `SpanContext`. And those are `default => sub{ random_bigint() }` (which is a wrapper)

if that is not the case (thanks to `should`) and not tested during runtime, 'someone' must have passed in the incorrect ID... which is afaik only possible at propagation time.

I rather see this fixed there than fix it afterwards with this bandage.

In other words, if Perl would have been a strongly typed language, this would never had occurred and raise other sorts of exceptions.

**Comment by vanHoesel** [comment]
Another, possible better place to fix this, is inside `SpanConntext` and coerce normal integers to `MathBigInt`. That way, any other that might want to extract ID's from a hash, will get the right result too.

**Comment by s-nez** [comment]
Yes, this approach is cleaner. I updated the attributes to coerce into BigInt and added modifiers for the clone methods to make sure their ids are also the right type (cloning doesn't go through constructors in `OpenTracing::Role::SpanContext`).

**Comment by vanHoesel** [comment]
Good catch!!
